src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -1040,6 +1064,40 @@ public bool ValidateTransaction(Transaction transaction, bool checkSignature = f
             }
         }
 
+        /// <inheritdoc />
+        public bool ValidateConsolidatingTransaction(Transaction transaction, bool checkSignature = false)
+        {
+            lock (this.lockObject)
+            {
+                List<Coin> coins = checkSignature ? new List<Coin>() : null;
+
+                // Verify that the transaction's UTXOs aren't used yet.
+                if (!this.TransactionIsSpendingUnspentUTXOs(transaction, coins))
+                    return false;
+
+                // TODO: Check the inputs are in order?
+
+                // Verify that all inputs are signed.
+                if (checkSignature)
+                {
+                    TransactionBuilder builder = new TransactionBuilder(this.Wallet.Network).AddCoins(coins);
+
+                    if (!builder.Verify(transaction, FederatedPegSettings.ConsolidationFee, out TransactionPolicyError[] errors))
+                    {
+                        // Trace the reason validation failed. Note that failure here doesn't mean an error necessarily. Just that the transaction is not fully signed.
+                        foreach (TransactionPolicyError transactionPolicyError in errors)
+                        {
+                            this.logger.LogDebug("TransactionBuilder.Verify FAILED - {0}", transactionPolicyError.ToString());
```

Comment:
You might want to `nameof` this or change the log cause of the `Verify` method name ever changes we will never know what this log referred to.